examples/evmapp/src/main/java/com/horizen/examples/EvmMessageProcessor.java

```diff
@@ -0,0 +1,45 @@
+package com.horizen.examples;
+
+import com.horizen.account.state.*;
+import com.horizen.evm.Evm;
+
+public class EvmMessageProcessor implements MessageProcessor {
+    @Override
+    public void init(AccountStateView view) {
+        // nothing to do here
+    }
+
+    /**
+     * Can process messages that require an EVM invocation:
+     * <ol>
+     * <li>when "to" address is empty; deploying a new contract</li>
+     * <li>when the account at the "to" address has non-empty code; invoke smart contract</li>
+     * </ol>
+     */
+    @Override
+    public boolean canProcess(Message msg, AccountStateView view) {
+        return msg.getTo() == null || view.getCodeHash(msg.getTo().address());
```

Comment:
`view.getCodeHash(msg.getTo().address())` can be replaced with `view.isSmartContractAccount(msg.getTo.address())` (was defined in the base branch recently)